SCClassLibrary/Common/Control/Buffer.sc

```diff
@@ -595,6 +601,16 @@ Buffer {
 		^[\b_query, bufnum]
 	}
 
+	setSampleRate { arg value;
+		server.listSendMsg(this.setSampleRateMsg(value));
+    }
+
+	setSampleRateMsg { arg value;
+		if(bufnum.isNil) { Error("Cannot change the sample rate of a % that has been freed".format(this.class.name)).throw };
+		if(value.isNil or: {value <= 0}) { sampleRate = server.sampleRate } {sampleRate = value.asFloat};
+		^[\b_setSampleRate, bufnum, value.asFloat];
```

Comment:
since we did all the checks and conversion on sampleRate, it would make sense to use it in the message:
```supercollider
^[\b_setSampleRate, bufnum, sampleRate];
```

Reply:
done!